src/Spice86/MemoryWrappers/MemoryBinaryDocument.cs

```diff
@@ -29,7 +34,7 @@ public class MemoryBinaryDocument : IBinaryDocument {
 
     public void ReadBytes(ulong offset, Span<byte> buffer) {
         for (int i = 0; i < buffer.Length; i++) {
-            buffer[i] = _memory[(uint)(offset + (uint)i)];
+            buffer[i] = _memory.UInt8[(int) _startAddress + (int)offset + i];
```

Comment:
Not really related to this change, but one thing to keep in mind is this will trigger read breakpoints. Maybe we need a way to tell the breakpoint system not to activate when components are used by the UI

Reply:
Good point. I think I need to use IMemory.GetSpan instead in order not to trigger them.

Reply:
This has been fixed by using GetSpan and adding an optional boolean parameter named triggerMemoryReadBreakpoints (default is true)